trieste/acquisition/function.py

```diff
@@ -544,161 +540,21 @@ def sample(self, at: TensorType) -> TensorType:
         produce the exact same samples. Calls to :meth:`sample` on *different*
         :class:`IndependentReparametrizationSampler` instances will produce different samples.
 
-        :param at: Where to sample the predictive distribution, with shape `[..., D]`, for points
+        :param at: Where to sample the predictive distribution, with shape `[..., 1, D]`, for points
             of dimension `D`.
-        :return: The samples, of shape `[..., S, L]`, where `S` is the `sample_size` and `L` is the
-            number of latent model dimensions.
-        :raise ValueError (or InvalidArgumentError): If ``at`` is a scalar.
+        :return: The samples, of shape `[..., S, 1, L]`, where `S` is the `sample_size` and `L` is
+            the number of latent model dimensions.
+        :raise ValueError (or InvalidArgumentError): If ``at`` has an invalid shape.
         """
-        tf.debugging.assert_rank_at_least(at, 1)
-        mean, var = self._model.predict(at[..., None, :])  # [..., 1, L], [..., 1, L]
+        tf.debugging.assert_shapes([(at, [..., 1, None])])
+        mean, var = self._model.predict(at[..., None, :, :])  # [..., 1, 1, L], [..., 1, 1, L]
 
         if tf.size(self._eps) == 0:
             self._eps.assign(
                 tf.random.normal([self._sample_size, mean.shape[-1]], dtype=tf.float64)
             )  # [S, L]
 
-        return mean + tf.sqrt(var) * tf.cast(self._eps, var.dtype)  # [..., S, L]
-
-
-class MCIndAcquisitionFunctionBuilder(AcquisitionFunctionBuilder):
```

Comment:
why is this not needed anymore?

Reply:
@joelberkeley ?

Reply:
this is scope creep. I can move it to a separate PR. Basically it hardly does anything and we were saying how it would be nice to have fewer classes

Reply:
yes, let's do that in a separate PR please!

Reply:
https://github.com/secondmind-labs/trieste/pull/210